### PR TITLE
WIP: vagrant-aws pipeline box provisioning

### PR DESCRIPTION
**Pull request by anguss00**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,3 +22,5 @@ private-sample/data_bags/passwords/s3_imos_restore.json
 private-sample/nodes/pipeline.json
 
 tmp
+.kitchen/
```
Comment (lwgordonimos) [inline .gitignore:25]: Is this left over from test kitchen prototyping?

File: chefignore [hunk 2]
```diff
@@ -0,0 +1 @@
+.kitchen
```
Comment (lwgordonimos) [inline chefignore:1]: testkitchen?

File: vagrant/aws.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    aws.instance_type = ENV['INSTANCE_TYPE']
+    aws.ami = ENV['SOURCE_AMI']
+    aws.security_groups = ["sg-0a7fce6c", "sg-0db869b77c7b4b418"]
+    aws.subnet_id = ENV['SUBNET_ID']
+    aws.keypair_name = ENV['SSH_KEY']
+    aws.elastic_ip = true
+    aws.associate_public_ip = true
+
+    override.ssh.username = "ec2-user"
+    override.ssh.private_key_path = "/home/grevious/.ssh/vagrant_rsa"
```
Comment (lwgordonimos) [inline vagrant/aws.rb:40]: This needs a sensible value, not sure what that is just with a cursory glance over this PR, but I think you mentioned the SSH stuff was still TODO.

File: vagrant/hash.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Hash
+  def slice(*keep_keys)
```
Comment (lwgordonimos) [inline vagrant/hash.rb:2]: Is this actually used anywhere in this PR?

File: Vagrantfile [hunk 5]
```diff
@@ -54,7 +46,7 @@ Vagrant.configure("2") do |config|
 
   define_node = Proc.new do |node_name|
     config.vm.define node_name, autostart: false do |node|
-      configure_virtualbox_provider node, node_name
+      configure_provider node, node_name
```
Comment (lwgordonimos) [inline Vagrantfile:49]: It seems like this configures *both* providers every time. Can it be refactored to only configure the provider it's actually about to use?

File: vagrant/aws.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  aws_init(ENV['AWS_PROFILE'])
+
+  config.vm.provider :aws do |aws, override|
+    aws.access_key_id = ENV['AWS_ACCESS_KEY_ID']
+    aws.secret_access_key = ENV['AWS_SECRET_ACCESS_KEY']
+    aws.session_token = ENV['AWS_SESSION_TOKEN']
+    aws.region = ENV['REGION']
+    aws.instance_type = ENV['INSTANCE_TYPE']
+    aws.ami = ENV['SOURCE_AMI']
+    aws.security_groups = ["sg-0a7fce6c", "sg-0db869b77c7b4b418"]
```
Comment (lwgordonimos) [inline vagrant/aws.rb:33]: Need to get rid of any static references to resources somehow. Needs to be totally dynamic/ephemeral.

File: Rakefile [hunk 7]
```diff
@@ -13,3 +13,10 @@ RSpec::Core::RakeTask.new(:spec) do |t, args|
   t.pattern = "cookbooks/*/spec/*_spec.rb"
   t.rspec_opts = "--color"
 end
+
+begin
```
Comment (lwgordonimos) [inline Rakefile:17]: More testkitchen

File: Vagrantfile (hunk 5 above)
Comment (anguss00) [reply Vagrantfile:49]: I believe that the aws specific provider will only run if --provider=aws is specified - the virtualbox one is the default

Comment (lwgordonimos) [reply Vagrantfile:49]: The provider will, but it looks like this calls both configure methods every time, and then obviously only *uses* one provider? https://github.com/aodn/chef/pull/1021/files#diff-8badd35a3a7b0e15d85fd51aeebca570R1

Comment (anguss00) [reply Vagrantfile:49]: Ah yeah gotcha. I think the stuff within the `config.vm.provider` block will only be executed if that provider is specified, but I see about that entry point. Should have a conditional there

**Pull request closed without merging**